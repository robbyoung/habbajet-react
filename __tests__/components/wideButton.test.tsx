import React from 'react';
import renderer from 'react-test-renderer';
import WideButton from '../../app/components/wideButton';
import {render, fireEvent, wait} from '@testing-library/react-native';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: '',
}));

describe('WideButton Component', () => {
    it('will render a button with specified color and text', () => {
        const component = renderer.create(
            <WideButton
                text="Test Button"
                color="#C2A066"
                onPress={() => undefined}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('can render an extra circle to hightlight the button', () => {
        const component = renderer.create(
            <WideButton
                text="Test Button"
                color="#DDDDDD"
                highlight={true}
                onPress={() => undefined}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('will run the onPress callback if the button is pressed', async () => {
        const onPress = jest.fn();
        const {getByTestId} = render(
            <WideButton
                text="Button"
                testID="test-button"
                color="#ffffff"
                onPress={onPress}
            />,
        );

        const button = getByTestId('test-button');
        fireEvent.press(button);
        await wait(() => expect(onPress).toBeCalled());
    });

    it('will not run the onPress callback if the button is disabled', async () => {
        const onPress = jest.fn();
        const {getByTestId} = render(
            <WideButton
                disabled={true}
                text="Button"
                testID="test-button"
                color="#ffffff"
                onPress={onPress}
            />,
        );

        const button = getByTestId('test-button');
        fireEvent.press(button);
        await wait(() => expect(onPress).not.toBeCalled());
    });
});
