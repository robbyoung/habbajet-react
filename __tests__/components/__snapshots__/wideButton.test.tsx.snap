// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WideButton Component will render a button with specified color and text 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#C2A066",
      "borderRadius": 10,
      "marginBottom": 10,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#ffffff",
          "fontFamily": "Abel",
          "fontSize": 30,
          "padding": 10,
          "textAlign": "center",
          "width": "100%",
        },
      ]
    }
  >
    Test Button
  </Text>
</View>
`;
